### PR TITLE
Update docker-compose test environment based on CI's tests

### PR DESCRIPTION
**Pull request by paulolobt**
Simulate the CI's tests with the docker-compose environment running in local. Adding the changes:
      - Add pip and curl Alpine packages on Dockerfile
      - Add env variable to run or not the tests
      - Create MariaDB and MySQL supervisor process
      - Create supervisor test config file
      - Run tests inside or not container
      - Run tests automatically on docker provision

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 FROM alpine:3.5
 
 # Install dependencies
-RUN apk add --no-cache php7-session php7-mysqli php7-mbstring php7-xml php7-gd php7-zlib php7-bz2 php7-zip php7-openssl php7-curl php7-opcache php7-json nginx php7-fpm supervisor
+RUN apk add --no-cache curl php7-session php7-mysqli php7-mbstring php7-xml php7-gd php7-zlib php7-bz2 php7-zip php7-openssl php7-curl php7-opcache php7-json nginx php7-fpm py2-pip supervisor
```
Comment (nijel) [inline Dockerfile]: This pulls whole python to the image, I'd prefer to avoid that.

File: etc/supervisord.conf [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+;priority=999                  ; the relative start priority (default 999)
+
+; The [include] section can just contain the "files" setting.  This
+; setting can list multiple files (separated by whitespace or
+; newlines).  It can also contain wildcards.  The filenames are
+; interpreted as relative to this file.  Included files *cannot*
+; include files themselves.
+
+[include]
+files = /etc/supervisor.d/nginx.ini /etc/supervisor.d/php.ini
```
Comment (nijel) [inline etc/supervisord.conf]: Couldn't this be handled somehow better? Having two almost same configuration files doesn't sound good for maintenance.

File: test-docker.sh [hunk 3]
```diff
@@ -1,47 +1,101 @@
...
+    $COMMAND_HOST cat /var/log/php-fpm.log
+    $COMMAND_HOST cat /var/log/nginx-error.log
+    $COMMAND_HOST cat /var/log/supervisord.log
+    echo -e "Result of ${PHPMYADMIN_DB_HOSTNAME} tests: ${RED}FAILED${NC}"
     exit $ret
 fi
 
-# List processes
-docker exec "$NAME" ps faux
+echo -e "Result of ${PHPMYADMIN_DB_HOSTNAME} tests: ${GREEN}SUCCESS${NC}"
```
Comment (nijel) [inline test-docker.sh]: The `echo -e` does not work in many /bin/sh implementations, it's better to avoid it (see Travis log for example).

**Comment by paulolobt** [comment]
Thanks your feedback @nijel. My mindset was only related to development mode and really the updates include unnecessary changes for production. So I will close this pull request and create a new with only docker-compose.testing.yml updates, because the compose test file not working. That's OK?

**Comment by nijel** [comment]
I've tried to base on your changes in https://github.com/phpmyadmin/docker/pull/111 (but it probably still needs some polishing).

**Comment by nijel** [comment]
I've based on your changes and implemented solution with additional Dockerfile to overlay the production one for testing. Also the docker-compose.testing.yml is now used during CI to ensure it's kept consistent with future changes.

**Pull request closed without merging**